File: id/src/main/java/com/simprints/id/activities/about/AboutContract.java
package com.simprints.id.activities.about;


import com.simprints.id.activities.BasePresenter;
import com.simprints.id.activities.BaseView;

interface AboutContract {

    interface View extends BaseView<Presenter> {

        void setVersionData(String appVersion, String libsimprintsVersion, String scannerVersion);

        void setDbCountData(String userCount, String moduleCount, String globalCount);

        void setRecoverDbAvailable();

        void setRecoverDbUnavailable();

        void setStartRecovering();

        void setSuccessRecovering();

        void setRecoveringFailed(String errorMessage);
    }

    interface Presenter extends BasePresenter {

        void recoverDb();
    }
}

File: id/src/main/java/com/simprints/id/activities/about/AboutContract.kt
package com.simprints.id.activities.about

import com.simprints.id.activities.BasePresenter
import com.simprints.id.activities.BaseView


interface AboutContract {

    interface View : BaseView<Presenter> {

        fun setVersionData(appVersion: String, libsimprintsVersion: String, scannerVersion: String)

        fun setUserCount(userCount: String)

        fun setProjectCount(projectCount: String)

        fun setModuleCount(moduleCount: String)

        fun setRecoveryInProgress()

        fun setSuccessRecovering()

        fun setRecoveringFailed()

    }

    interface Presenter : BasePresenter {

        fun recoverDb()

    }
}
